### PR TITLE
VTCCA-4622 When you became the owner or occupier and Do you still own property pages

### PR DESCRIPTION
**Pull request by reddy-yattapu**
https://jira.tools.tax.service.gov.uk/browse/VTCCA-4622

### CODE REVIEW

File: app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import services.BusinessRatesAttachmentsService
+import services.propertylinking.PropertyLinkingService
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import uk.gov.hmrc.propertylinking.errorhandler.CustomErrorHandler
+import uk.gov.voa.play.form.ConditionalMappings._
+import views.helpers.Errors
+import java.time.LocalDate
+
+import javax.inject.{Inject, Named}
+import org.checkerframework.checker.compilermsgs.qual.CompilerMessageKey
```
Comment (darrell-plant) [inline app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala:43]: what is this?

File: app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+@Singleton
+class ClaimPropertyOccupancyController @Inject()(
+      val errorHandler: CustomErrorHandler,
+      val submissionIdConnector: SubmissionIdConnector,
+      @Named("propertyLinkingSession") val sessionRepository: SessionRepo,
+      authenticatedAction: AuthenticatedAction,
+      withLinkingSession: WithLinkingSession,
+      val propertyLinksConnector: PropertyLinkConnector,
+      businessRatesAttachmentService: BusinessRatesAttachmentsService,
+      val runModeConfiguration: Configuration,
```
Comment (darrell-plant) [inline app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala:57]: is this used?

File: app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+          businessRatesAttachmentService
+            .persistSessionData(request.ses.copy(
+              propertyOccupancy = Some(PropertyOccupancy(stillOccupied = true, lastOccupiedDate = None))))
+            .map { _ =>
+              Redirect(routes.ChooseEvidenceController.show())
+            }
+            .recover {
+              case UpstreamErrorResponse.Upstream5xxResponse(_) =>
+                ServiceUnavailable(serviceUnavailableView())
+            }
```
Comment (darrell-plant) [inline app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala:83]: do we need these `recovers` ?

File: app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+import javax.inject.{Inject, Named}
+import org.checkerframework.checker.compilermsgs.qual.CompilerMessageKey
+import utils.Formatters
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ClaimPropertyOccupancyController @Inject()(
+      val errorHandler: CustomErrorHandler,
```
Comment (PeterPerhac) [inline app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala:50]: this controller seems to have a ton of things injected into its constructor. Does it really need all this?
```
    val submissionIdConnector: SubmissionIdConnector,
    val propertyLinksConnector: PropertyLinkConnector,
    val runModeConfiguration: Configuration,
```
 can be removed

**Comment by darrell-plant** [comment]
LGTDz 👍 (minor comments)

File: app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+      propertyLinkingService.findEarliestStartDate(request.ses.uarn).flatMap {
+        case Some(date) if date.isAfter(LocalDate.now()) =>
+          businessRatesAttachmentService
+            .persistSessionData(request.ses.copy(
+              propertyOccupancy = Some(PropertyOccupancy(stillOccupied = true, lastOccupiedDate = None))))
+            .map { _ =>
+              Redirect(routes.ChooseEvidenceController.show())
+            }
+            .recover {
```
Comment (PeterPerhac) [inline app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala:80]: do we need to recover with a service unavailable view? I think if there's a upstream 5xx response, can't we let that failed future be handled by global error handler? it just displays service unavailable anyways

File: app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+                  errors,
+                  request.ses.clientDetails,
+                  controllers.propertyLinking.routes.ClaimPropertyOwnershipController.showOwnership().url))),
+          formData =>
+            businessRatesAttachmentService
+              .persistSessionData(request.ses.copy(propertyOccupancy = Some(formData)))
+              .map { _ =>
+                Redirect(routes.ChooseEvidenceController.show())
+              }
+              .recover {
```
Comment (PeterPerhac) [inline app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala:121]: same here, i don't think we need to recover like this. let it fail and error handler will display service unavailable

**Comment by darrell-plant** [comment]
👍

**Pull request merged**